### ISSUE TITLE
Callbacks in `HasManyRepeater` schemas receive the top-level record when creating new items

### ISSUE THREAD

**Issue by jacksleight** [laravel-filament/filament#1563]
### Package

filament/filament

### Package Version

v2.10.5

### Laravel Version

v9.0.1

### Livewire Version

v2.10.1

### PHP Version

8.0.14

### Bug description

I have a `Snag` model with a `comments` HasMany relation to a `SnagComment` model.

In my `SnagResource` I have the following (simplified) schema, which outputs the comments in a `HasManyRepeater`, displaying the `created_at` date and content field:

```php
class SnagResource extends Resource
{
    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\HasManyRepeater::make('comments')
                    ->relationship('comments')
                    ->schema([
                        Forms\Components\Placeholder::make('created_at')
                            ->content(fn (?SnagComment $record): string => $record ? $record->created_at->diffForHumans() : 'Now'),
                        Forms\Components\Textarea::make('content'),
                    ]),
                ]);
    }
}
```

The `created_at` content callback expects a `SnagComment` model, or nothing at all if it's a new item in the repeater. However when I create a new item I get the following error:

> App\Filament\Resources\SnagResource::App\Filament\Resources\{closure}(): Argument 1 ($record) must be of type ?App\Models\SnagComment, App\Models\Snag given

The callback is receiving the top-level `Snag` model, not an empty value as (I think?) it should.

Existing items in the repeater work fine, the callbacks receive the `SnagComment` model as expected.

### Steps to reproduce

Create a `HasManyRepeater` field that uses callbacks that expect an instance of the model being repeated.

**Comment by jacksleight** [comment]
It feels like that should work, but for some reason it doesn't. If I have this:

```php
Forms\Components\Placeholder::make('created_at')
    ->content(fn ($state): string => $state ? Carbon::parse($state)->created_at->toDayDateTimeString() : 'Now'),
```

`$state` is always `null`, even for existing records. Maybe I've done something wrong?

Although this raises another question. In my actual schema I'm doing this:

```php
Forms\Components\HasManyRepeater::make('comments')
    ->relationship('comments')
    ->schema([
        Forms\Components\Textarea::make('content')
            ->label(fn (?SnagComment $record): string => $record ? $record->created_at->toDayDateTimeString() : 'Now')
            ->required(),
    ]),
```

I'm trying to use the comment date as the label for the text field (to simplify the UI). But in that instance `$state` is the comment text. Ideally I need access to all of the comment attributes from any field still.

**Comment by danharrin** [comment]
Should be fixed by #1574?

**Comment by jacksleight** [comment]
> Should be fixed by #1574?

Yup working fine now 👍
